Foundation/JSONSerialization.swift

```diff
@@ -28,6 +28,253 @@ extension JSONSerialization {
     }
 }
 
+fileprivate protocol _JSONValue {
+    func visit<Visitor: JSONVisitor>(with visitor: inout Visitor) throws
+}
+
+//@_specialize(where T == String)
+//extension<T: _JSONValue> [String : T] : _JSONValue {
+extension Dictionary : _JSONValue where Key == String, Value : _JSONValue {
+    
+//    @_specialize(where Visitor == JSONWriter)
+//    @_specialize(where Visitor == JSONValidator)
```

Comment:
@_specialize isn't clever enough to recover the Key and Value constraints from the where clause above, and also doesn't support the 'Value : _JSONValue' constraint currently

Reply:
Is there somewhere I could read about this new feature?

Reply:
We chatted on slack about this